enigma-tools-u/src/esgx/general.rs

```diff
@@ -1,52 +1,54 @@
 use sgx_types::{sgx_attributes_t, sgx_launch_token_t, sgx_misc_attribute_t, SgxResult};
 use sgx_urts::SgxEnclave;
-use std::env;
-use std::fs;
-use std::io::Read;
-use std::path::{self, PathBuf, Path};
-use dirs;
+use std::{fs, io::Read, path, io::Write};
+use std::path::{PathBuf, Path};
 use failure::Error;
 
 pub fn storage_dir<P: AsRef<Path>>(dir_name: P) -> Result<PathBuf, Error> {
-    let mut path = dirs::home_dir().ok_or_else(|| format_err!("Missing HomeDir"))?;
+    let mut path = dirs::home_dir().ok_or_else(|| {
+        format_err!("Missing home directory")
+    })?;
     trace!("Home dir is {}", path.display());
     path.push(dir_name);
     Ok(path)
 }
 
-pub fn init_enclave(token_path: &path::PathBuf, use_token: bool, enclave_location: &str)
-    -> SgxResult<(SgxEnclave, Option<sgx_launch_token_t>)> {
-    let path = env::current_dir().unwrap();
-    trace!("The current directory is {}", path.display());
+pub fn init_enclave(token_path: &path::PathBuf, enclave_location: &str)
+    -> SgxResult<(SgxEnclave)> {
     let mut launch_token: sgx_launch_token_t = [0; 1024];
     let mut launch_token_updated: i32 = 0;
 
-    if use_token {
-        match fs::File::open(&token_path) {
-            Err(_) => {
-                error!("Open token file {} error! Will create one.", token_path.as_path().to_str().unwrap());
+    match fs::File::open(&token_path) {
+        Err(e) => {
+            match e.kind() {
+                std::io::ErrorKind::NotFound => info!("Enclave launch token does not exist. Will create one."),
+                _ => warn!("Cannot open enclave launch token file: {}.", e)
             }
-            Ok(mut f) => {
-                info!("Open token file success! ");
-                match f.read(&mut launch_token) {
-                    Ok(1024) => {
-                        info!("Token file valid!");
-                    }
-                    _ => warn!("Token file invalid, will create new token file"),
-                }
+        },
+        Ok(mut f) => {
+            match f.read(&mut launch_token) {
+                Ok(1024) => info!("Enclave launch token found."),
+                _ => warn!("Enclave launch token invalid, will create a new one."),
             }
         }
     }
 
-    // Step 2: call sgx_create_enclave to initialize an enclave instance
+    // Call sgx_create_enclave to initialize an enclave instance
     // Debug Support: set 2nd parameter to 1
     let debug = 1;
     let mut misc_attr = sgx_misc_attribute_t { secs_attr: sgx_attributes_t { flags: 0, xfrm: 0 }, misc_select: 0 };
     let enclave = SgxEnclave::create(enclave_location, debug, &mut launch_token, &mut launch_token_updated, &mut misc_attr)?;
```

Comment:
can you please document this function- explain that the token is created in here if it didn't exist until now. maybe add an explanation of the updated flag as well. thanks!

Reply:
I removed all the **token** related stuff, since it is deprecated https://github.com/apache/incubator-teaclave-sgx-sdk/pull/163